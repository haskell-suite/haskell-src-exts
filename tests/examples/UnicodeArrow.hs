{-# LANGUAGE UnicodeSyntax #-}

arrowTest ∷ String →   String
arrowTest input =
    case input of
      "hello" → "world"
      "world" → "hello"
      otherwise → "unknow"
