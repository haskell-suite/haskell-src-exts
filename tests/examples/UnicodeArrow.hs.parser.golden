ParseOk (Module (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 1 1 9 1, srcInfoPoints = [SrcSpan "tests/examples/UnicodeArrow.hs" 1 1 1 1,SrcSpan "tests/examples/UnicodeArrow.hs" 3 1 3 1,SrcSpan "tests/examples/UnicodeArrow.hs" 3 1 3 1,SrcSpan "tests/examples/UnicodeArrow.hs" 3 1 3 1,SrcSpan "tests/examples/UnicodeArrow.hs" 4 1 4 1,SrcSpan "tests/examples/UnicodeArrow.hs" 9 1 9 1,SrcSpan "tests/examples/UnicodeArrow.hs" 9 1 9 1]}) Nothing [LanguagePragma (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 1 1 1 31, srcInfoPoints = [SrcSpan "tests/examples/UnicodeArrow.hs" 1 1 1 13,SrcSpan "tests/examples/UnicodeArrow.hs" 1 28 1 31]}) [Ident (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 1 14 1 27, srcInfoPoints = []}) "UnicodeSyntax"]] [] [TypeSig (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 3 1 3 32, srcInfoPoints = [SrcSpan "tests/examples/UnicodeArrow.hs" 3 11 3 13]}) [Ident (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 3 1 3 10, srcInfoPoints = []}) "arrowTest"] (TyFun (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 3 14 3 32, srcInfoPoints = [SrcSpan "tests/examples/UnicodeArrow.hs" 3 21 3 23]}) (TyCon (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 3 14 3 20, srcInfoPoints = []}) (UnQual (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 3 14 3 20, srcInfoPoints = []}) (Ident (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 3 14 3 20, srcInfoPoints = []}) "String"))) (TyCon (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 3 26 3 32, srcInfoPoints = []}) (UnQual (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 3 26 3 32, srcInfoPoints = []}) (Ident (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 3 26 3 32, srcInfoPoints = []}) "String")))),FunBind (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 4 1 8 28, srcInfoPoints = []}) [Match (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 4 1 8 28, srcInfoPoints = []}) (Ident (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 4 1 4 10, srcInfoPoints = []}) "arrowTest") [PVar (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 4 11 4 16, srcInfoPoints = []}) (Ident (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 4 11 4 16, srcInfoPoints = []}) "input")] (UnGuardedRhs (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 4 17 8 28, srcInfoPoints = [SrcSpan "tests/examples/UnicodeArrow.hs" 4 17 4 18]}) (Case (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 5 5 8 28, srcInfoPoints = [SrcSpan "tests/examples/UnicodeArrow.hs" 5 5 5 9,SrcSpan "tests/examples/UnicodeArrow.hs" 5 16 5 18,SrcSpan "tests/examples/UnicodeArrow.hs" 6 7 6 7,SrcSpan "tests/examples/UnicodeArrow.hs" 7 7 7 7,SrcSpan "tests/examples/UnicodeArrow.hs" 8 7 8 7,SrcSpan "tests/examples/UnicodeArrow.hs" 9 1 9 0]}) (Var (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 5 10 5 15, srcInfoPoints = []}) (UnQual (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 5 10 5 15, srcInfoPoints = []}) (Ident (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 5 10 5 15, srcInfoPoints = []}) "input"))) [Alt (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 6 7 6 25, srcInfoPoints = []}) (PLit (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 6 7 6 14, srcInfoPoints = []}) (Signless (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 6 7 6 14, srcInfoPoints = []})) (String (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 6 7 6 14, srcInfoPoints = []}) "hello" "hello")) (UnGuardedRhs (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 6 15 6 25, srcInfoPoints = [SrcSpan "tests/examples/UnicodeArrow.hs" 6 15 6 17]}) (Lit (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 6 18 6 25, srcInfoPoints = []}) (String (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 6 18 6 25, srcInfoPoints = []}) "world" "world"))) Nothing,Alt (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 7 7 7 25, srcInfoPoints = []}) (PLit (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 7 7 7 14, srcInfoPoints = []}) (Signless (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 7 7 7 14, srcInfoPoints = []})) (String (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 7 7 7 14, srcInfoPoints = []}) "world" "world")) (UnGuardedRhs (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 7 15 7 25, srcInfoPoints = [SrcSpan "tests/examples/UnicodeArrow.hs" 7 15 7 17]}) (Lit (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 7 18 7 25, srcInfoPoints = []}) (String (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 7 18 7 25, srcInfoPoints = []}) "hello" "hello"))) Nothing,Alt (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 8 7 8 28, srcInfoPoints = []}) (PVar (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 8 7 8 16, srcInfoPoints = []}) (Ident (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 8 7 8 16, srcInfoPoints = []}) "otherwise")) (UnGuardedRhs (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 8 17 8 28, srcInfoPoints = [SrcSpan "tests/examples/UnicodeArrow.hs" 8 17 8 19]}) (Lit (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 8 20 8 28, srcInfoPoints = []}) (String (SrcSpanInfo {srcInfoSpan = SrcSpan "tests/examples/UnicodeArrow.hs" 8 20 8 28, srcInfoPoints = []}) "unknow" "unknow"))) Nothing])) Nothing]],[])
